### PR TITLE
Cortex Attack Surface Management 2.0.0

### PR DESCRIPTION
**Pull request by johnnywilkes**
## Contributing to Cortex XSOAR Content
Make sure to register your contribution by filling the [contribution registration form](https://forms.gle/XDfxU4E61ZwEESSMA)

**The Pull Request will be reviewed only after the contribution registration form is filled.**

## Status
- [x] In Progress
- [ ] Ready
- [ ] In Hold - (Reason for hold)

## Related Issues
fixes: link to the issue

## Description
Consider this a re-imaging of the "Cortex ASM - ASM Alert" playbook to make it useful to handle ASM related alerts.  Please let us know if we should submit a demo video or if you would like to do it live!

## Screenshots
Paste here any images that will help the reviewer

## Minimum version of Cortex XSOAR
- [ ] 6.0.0
- [ ] 6.1.0
- [ ] 6.2.0
- [X] 6.5.0

## Does it break backward compatibility?
   - [ ] Yes
       - Further details:
   - [X] No

## Must have
- [X] Tests
- [X] Documentation

### CODE REVIEW

**Comment by johnnywilkes** [comment]
@ShacharKidor , are you able to take look at this, please?

**Comment by ShacharKidor** [comment]
> @ShacharKidor / @idovandijk / @altmannyarden are you able to take look at this, please?

Hi @johnnywilkes - a reviewer from our contribution team will start the review soon. 

@dorschw - FYI.

**Review by idovandijk — COMMENTED** [review]
Thank you for yet another contribution @johnnywilkes, your time and effort are greatly appreciated.
Here are my comments for the playbooks:

### Cortex ASM - Detect Service
1. Should this be OR or AND? The operator currently applied is "and" but the title says "or". Also later on in the Cortex ASM - ASM Alert playbook, there is a task that sets the value of "None" if there is no remote IP, which suggests to me that there is not always an IP in the alert:
image
image


### Cortex ASM - ASM Alert
2. Question - is there a scenario where this grid field gets more than one entry? If so, that can interfere with this condition:
image
If the answer to the question is yes, you may want to consider one of the following changes:
A. Using a playbook output to determine if the port is closed, rather than checking the grid field values.
B. Changing the condition to "GET `alert.asmservicedetection` WHERE `alert.asmservicedetection.ScanResult` DOESN'T EQUAL `Closed` - IS DEFINED" - which would check if any entry has a value other than "closed".
If there will always be one entry there, then no change is required here.
3. What data does the "ASM Tags" field hold? Why does the occurrence of "dev" imply that automated remediation can be done? E.g., is this something that should be configurable?
image

Please let me know when we can do a short demo for this. Feel free to reach out if you have any questions.

File: Packs/CortexAttackSurfaceManagement/Scripts/GenerateASMReport/GenerateASMReport_test.py [hunk 1]
```diff
@@ -0,0 +1,72 @@
...
+from CommonServerPython import EntryType
+
+
+def util_load_json(path):
+    with open(path, mode="r") as f:
+        return json.loads(f.read())
+
+
+def test_get_asm_args(mocker):
+    from GenerateASMReport import get_asm_args
```
[the remark this answers is not in the document]
Comment (troberts13) [reply Packs/CortexAttackSurfaceManagement/Scripts/GenerateASMReport/GenerateASMReport_test.py:12]: @dorschw  where is this documented for reference? I don't see it mentioned here: https://xsoar.pan.dev/docs/integrations/unit-testing

File: Packs/CortexAttackSurfaceManagement/Scripts/GetRemediationGuidance/GetRemediationGuidance.py [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+
+
+def get_remediation_guidance_command(args: Dict[str, Any]) -> CommandResults:
+
+    ASM_rule_ID = args.get('AttackSurfaceRuleID')
+    result = get_remediation_guidance(str(ASM_rule_ID))
+    markdown = tableToMarkdown('Remediation Guidance', result, headers=['Guidance'])
+    return CommandResults(
+        outputs_prefix='RemediationGuidance',
+        outputs_key_field='',
```
[the remark this answers is not in the document]
Comment (johnnywilkes) [reply Packs/CortexAttackSurfaceManagement/Scripts/GetRemediationGuidance/GetRemediationGuidance.py]: we are replacing that automation withe a playbook/list (Cortex ASM - Remediation Guidance/RemediationGuidance) so I don't need to make a change here

File: Packs/CortexAttackSurfaceManagement/Scripts/GenerateASMReport/GenerateASMReport.yml [hunk 3]
```diff
@@ -0,0 +1,56 @@
...
+  name: alert_id
+- description: Name of the Alert that triggered this playbook
+  name: alert_name
+- description: Details of the Alert that triggered this playbook
+  name: alert_details
+- description: ASM Alert Severity String
+  name: alert_severityStr
+- description: Potential service owners gathered through playbook.
+  isArray: true
+  name: asmserviceowner
```
[the remark this answers is not in the document]
Comment (johnnywilkes) [reply Packs/CortexAttackSurfaceManagement/Scripts/GenerateASMReport/GenerateASMReport.yml]: @dorschw , changes we made:
asmserviceowner -> asm_service_owner
asmremediation -> asm_remediation
asmservicedetection -> asm_service_detection
asmsystemids -> asm_system_ids
asmcloud -> asm_cloud
asmnotification -> asm_notification
asmdatacollection -> asm_data_collection
asmtags -> asm_tags
asmprivateip -> asm_private_ip
asmrelated -> asm_related

File: Packs/CortexAttackSurfaceManagement/Scripts/GenerateASMReport/GenerateASMReport_test.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (johnnywilkes) [reply Packs/CortexAttackSurfaceManagement/Scripts/GenerateASMReport/GenerateASMReport_test.py:21]: @dorschw , we have added these, let us know what you think and if we should add some type of documentation to: https://xsoar.pan.dev/docs/integrations/code-conventions#unit-tests

**Comment by johnnywilkes** [comment]
@idovandijk / @dorschw 
We have made all changes we talked about Thursday. In Summary:

Ido's #1 comment:
- In "Cortex ASM - Detect Service" we changed the first conditional to "and" instead of "or" (title and description also)
image
- In "Cortex ASM - Enrichment" we made RemoteIP input no longer mandatory and simplified the first conditional of the playbook (removed 2nd expression):
image
- In "Cortex ASM - ASM Alert" we removed the "SetIfEmpty" transform for the "Cortex ASM - Enrichment" subplaybook as it was no longer needed:
image

Ido's #2 comment: We verified with our engineering group that ${alert.RemoteIP} and ${alert.RemotePort} should only be at most a single entry.

Per @dorschw 's suggestion, we removed the "GetRemediationGuidance" automation and replaced it with a new subplaybook (Cortex ASM - Remediation Guidance) and list (RemediationGuidance).  @idovandijk , i am assuming you will have to look over that playbook as well.

Also changed inputs and adding unit-test docstrings to GenerateASMReport automation.


Please let me know if you have any comments, questions or concerns.  Thank you again for your assistance.

File: Packs/CortexAttackSurfaceManagement/Scripts/GenerateASMReport/GenerateASMReport_test.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (dorschw) [reply Packs/CortexAttackSurfaceManagement/Scripts/GenerateASMReport/GenerateASMReport_test.py:21]: thanks for the suggestion, it should be fixed in [this PR](https://github.com/demisto/content-docs/pull/1223)

File: Packs/CortexAttackSurfaceManagement/ReleaseNotes/2_0_0.md [hunk 4]
```diff
@@ -0,0 +1,35 @@
+
+#### Playbooks
+
+##### Cortex ASM - ASM Alert
+Replaced earlier **Cortex ASM - ASM Alert** playbook and Readme files.
```
Comment (dorschw) [inline Packs/CortexAttackSurfaceManagement/ReleaseNotes/2_0_0.md:5]: what is changed in the playbook? What difference should users expect?

File: Packs/CortexAttackSurfaceManagement/pack_metadata.json [hunk 5]
```diff
@@ -2,7 +2,7 @@
     "name": "Cortex Attack Surface Management",
     "description": "Content for working with Attack Surface Management (ASM).",
     "support": "xsoar",
-    "currentVersion": "1.1.0",
+    "currentVersion": "2.0.0",
```
Comment (dorschw) [inline Packs/CortexAttackSurfaceManagement/pack_metadata.json:5]: Major version bumps usually mark breaking changes, or significant changes - is this the case here? If it's a feature and backwards compatible, I'd change it to a minor bump (1.1.0 -> 1.2.0)

**Review by dorschw — COMMENTED** [review]
Two minor comments, aside from this, the PR is approved on my end.

**Comment by ShirleyDenkberg** [comment]
@idovandijk @troberts13 @dorschw Doc review completed.

**Review by idovandijk — COMMENTED** [review]
Reviewed the recent playbook changes, looks great!
@dorschw feel free to approve this PR once you're done with the code review.

File: Packs/CortexAttackSurfaceManagement/ReleaseNotes/2_0_0.md (hunk 4 above)
Comment (johnnywilkes) [reply Packs/CortexAttackSurfaceManagement/ReleaseNotes/2_0_0.md:5]: @dorschw , the original playbook was created really just as an "example" and not sure it was actually used.  It focused on enrichment whereas now there are additional stages (triage, enrichment, decision, remediation, ticketing, validation, summarization).  Let me know if you want me to add more data in this release note.

File: Packs/CortexAttackSurfaceManagement/pack_metadata.json (hunk 5 above)
Comment (johnnywilkes) [reply Packs/CortexAttackSurfaceManagement/pack_metadata.json:5]: @dorschw , as mentioned in the previous comment there were some significant changes in the main playbook.  I don't really consider them "breaking" as the previous playbook was just setup as and example, and don't believe it was being used.  That being said, I don't mind using any version number that you recommend.  Let me know what change (if any) I should make, please.

File: Packs/CortexAttackSurfaceManagement/ReleaseNotes/2_0_0.md (hunk 4 above)
Comment (dorschw) [reply Packs/CortexAttackSurfaceManagement/ReleaseNotes/2_0_0.md:5]: Please mention the additional stages (`The playbook now allows ...`)

File: Packs/CortexAttackSurfaceManagement/pack_metadata.json (hunk 5 above)
Comment (dorschw) [reply Packs/CortexAttackSurfaceManagement/pack_metadata.json:5]: let's do `1.2.0`, just change the release note file name, and the version value under `pack_metadata.json`, and we'll be ready to merge it :)

Comment (johnnywilkes) [reply Packs/CortexAttackSurfaceManagement/pack_metadata.json]: @dorschw , just pushed changes

**Review by dorschw — APPROVED** [review]
🎉

**Pull request merged**